includes/Core/Remote_Features/Remote_Features_Provider.php

```diff
@@ -93,6 +102,30 @@ public function register() {
 		$this->cron->register();
 
 		add_action( 'admin_init', fn () => $this->on_admin_init() );
+		add_action(
+			'heartbeat_tick',
+			fn ( $response, $screen_id ) => $this->on_heartbeat_tick( $response, $screen_id ),
+			10,
+			2
+		);
+	}
+
+	/**
+	 * Handles the heartbeat AJAX callback.
+	 *
+	 * @param array  $response  The Heartbeat response.
+	 * @param string $screen_id The screen ID.
+	 */
+	protected function on_heartbeat_tick( $response, $screen_id ) {
+		if ( ! $this->credentials->has() || ! $this->credentials->using_proxy() ) {
+			return;
+		}
```

Comment:
These checks on `credentials` aren't needed since the same guards are applied on the `syncer` itself. This was more important before when we were going to guard the self-request, but since this is contained within the existing heartbeat call, then it isn't really necessary.

As an aside, it's always preferable to check simple guards before more complex ones – i.e. the simple `screen_id` check below would be better to make before making calls that will load encrypted `credentials`, etc.